Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -139,4 +160,39 @@ public class Channel {
       observer.addConnectivityObserver(callback: callback)
     }
   }
+
+  // MARK: - Private
+
+  private init(host: String, underlyingChannel: UnsafeMutableRawPointer) {
+    self.host = host
+    self.underlyingChannel = underlyingChannel
+    self.completionQueue = CompletionQueue(underlyingCompletionQueue: cgrpc_channel_completion_queue(underlyingChannel),
+                                           name: "Client")
+
+    self.completionQueue.run()
+    self.scheduleActiveCallCleanUp()
+  }
+
+  private func scheduleActiveCallCleanUp() {
+    DispatchQueue.global(qos: .background).asyncAfter(deadline: .now() + 10.0) { [weak self] in
+      self?.cleanUpActiveCalls()
+    }
+  }
+
+  private func cleanUpActiveCalls() {
+    self.mutex.synchronize {
+      self.activeCalls = self.activeCalls.filter { $0.value != nil }
+    }
+    self.scheduleActiveCallCleanUp()
+  }
+}
+
+/// Used to hold weak references to objects since `NSHashTable<T>.weakObjects()` isn't available on Linux.
+/// If/when this type becomes available on Linux, this should be replaced.
+private final class WeakReference<T: AnyObject> {
+  weak var value: T?
```

Comment:
`private(set) weak`, to make it more clear that this can only be changed by a deallocation?

Reply:
good call